algo/src/main/java/com/hartwig/serve/sources/ckb/blacklist/CkbStudyBlacklistModel.java

```diff
@@ -0,0 +1,118 @@
+package com.hartwig.serve.sources.ckb.blacklist;
+
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.collect.Sets;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+
+public class CkbStudyBlacklistModel {
+
+    private static final Logger LOGGER = LogManager.getLogger(CkbStudyBlacklistModel.class);
+
+    @NotNull
+    private final List<CkbBlacklistStudyEntry> blacklistStudiesList;
+    @NotNull
+    private final Set<CkbBlacklistStudyEntry> usedBlacklists = Sets.newHashSet();
+
+    public CkbStudyBlacklistModel(@NotNull final List<CkbBlacklistStudyEntry> blacklistStudiesList) {
+        this.blacklistStudiesList = blacklistStudiesList;
+    }
+
+    public boolean isBlacklistStudy(@NotNull String studyName, @NotNull String therapyName, @NotNull String cancerType, @NotNull String sourceGene,
+                                    @NotNull String event) {
+        for (CkbBlacklistStudyEntry blacklistStudyEntry : blacklistStudiesList) {
+            boolean match = isMatch(studyName, therapyName, cancerType, sourceGene, event, blacklistStudyEntry);
+            if (match) {
+                usedBlacklists.add(blacklistStudyEntry);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public void reportUnusedBlacklistEntries() {
+        int unusedBlacklistEntryCount = 0;
+        for (CkbBlacklistStudyEntry entry : blacklistStudiesList) {
+            if (!usedBlacklists.contains(entry)) {
+                unusedBlacklistEntryCount++;
+                LOGGER.warn(" Blacklist entry '{}' hasn't been used for CKB filtering", entry);
+            }
+        }
+        LOGGER.debug(" Found {} unused blacklist entries during CKB filtering", unusedBlacklistEntryCount);
+    }
+
+    public boolean isMatch(@NotNull String studyName, @NotNull String therapyName, @NotNull String cancerType, @NotNull String sourceGene,
+                           @NotNull String event, @NotNull CkbBlacklistStudyEntry blacklistStudyEntry) {
+        switch (blacklistStudyEntry.type()) {
+            case STUDY_WHOLE: {
```

Comment:
You actually filter on `nctId` rather than `studyName`, does that make sense? May benefit from clearer naming?

Reply:
Variable name was unclear...it was actually already the nctId and not the studyName